chat_app_be/chat_app_be/Controllers/AuthController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using chat_app_be.Dtos;
+using chat_app_be.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using chat_app_be.Interfaces;
+using chat_app_be.Models.Auth;
+
+namespace chat_app_be.Controllers
+{
+    [Route("api/")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly JwtOptions _jwtOptions;
+        private readonly IUserRepository _userRepository;
+
+        public AuthController(IOptions<JwtOptions> jwtOptions, IUserRepository userRepository)
+        {
+            _jwtOptions = jwtOptions.Value;
+            _userRepository = userRepository;
+        }
+
+        [HttpPost("register")]
+        public async Task<ActionResult<UserRequestDto>> Register(UserDto user)
+        {
+            if (await _userRepository.UserExistsAsync(user.Username))
```

Comment:
I would recommend separating the business logic into the Service layer to enhance your application's maintainability, scalability, and testability.

You can read more at https://learn.microsoft.com/en-us/aspnet/mvc/overview/older-versions-1/models-data/validating-with-a-service-layer-cs#creating-a-service-layer